### PR TITLE
WTD: projected volumes

### PR DESCRIPTION
**Pull request by jseldess**
- Add projected volume type to the volumes concept page:

  I've kept this light since I don't fully understand the feature, 
  but since this is a distinct volume type, it seems helpful to
  list it here. Also, this lets us keep concept details out of the
  related task, and it gives us a good place to link to the original
  design document.

- Apply template to projected volume task:

  This involves incorporating an actual scenario and series
  of steps. I don't think the configuration is correct,
  but I'm hoping the general trajectory is moving in the right
  direction.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3771)

### CODE REVIEW

**Comment by chenopis** [comment]
I'll have an engineer to a tech review.

**Comment by jseldess** [comment]
@saad-ali, just a few extra details:

- The pod config I used is a combination of the one in the basic volume tutorial [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-volume-storage/#configuring-a-volume-for-a-pod) and configs in the projected volume design doc [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/all-in-one-volume.md). I doubt this is the right approach, but I just wanted to get a basic example put together.

- Also, given that the example projects secrets, I'm not sure how realistic it is to tell users to exec into the container and list the contents of the `projected-volume` directory.

**Comment by jseldess** [comment]
@jpeeler and/or @pmorie as well, since you are listed as assignees in the projected volumes file.

File: docs/tasks/configure-pod-container/projected-volume.md [hunk 1]
```diff
@@ -8,35 +8,41 @@ redirect_from:
...
-individual type. Consider the following example:
+{% include task-tutorial-prereqs.md %}
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Configure a pod to project multiple sources into a single directory
+
+In this exercise, you create a Pod that runs Redis in one Container. The Pod uses two types of Volumes: [`emptyDir`](/docs/concepts/storage/volumes/#emptydir) is used for storing Redis data, and [`projected`](/docs/concepts/storage/volumes/#projected) mounts multiple secrets into the same shared directory. Here is the configuration file for the Pod:
```
Comment (jpeeler) [inline docs/tasks/configure-pod-container/projected-volume.md]: I'm on the fence about adding anything about Redis, but perhaps it makes the scenario more realistic. In such a case, I think it's best to mention creating the secrets is a prerequisite for the pod to start.

File: docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -553,9 +554,9 @@ __Important: Make sure you have an existing PortworxVolume with name `pxvol` bef
 More details and examples can be found [here](https://github.com/kubernetes/kubernetes/tree/{{page.githubbranch}}/examples/volumes/portworx/README.md)
 
 ### ScaleIO
-ScaleIO is a software-based storage platform that can use existing hardware to create clusters of scalable
```
Comment (jpeeler) [inline docs/concepts/storage/volumes.md]: I'm not sure how strict k8s is about separating pull requests, but these formatting changes in my opinion should be in a separate commit at least (if not an entirely separate PR).

Comment (jseldess) [inline docs/concepts/storage/volumes.md]: Sorry about that. My text editor is just set up to trim whitespace at the end of lines. No change to the format of the rendered html, though.

File: docs/tasks/configure-pod-container/projected-volume.md (hunk 1 above)
Comment (jpeeler) [inline docs/tasks/configure-pod-container/projected-volume.md]: Ah, you're right. The busybox scenario I always use is missing the args, should look like:
```
image: busybox
args:
- sleep
- "86400"
```
And yes, I think adding steps to create the secrets would be a nice addition.

**Comment by mburke5678** [comment]
@jseldess @jpeeler Can you please let me know when this document is complete so I can bring it downstream into the OpenShift docs.

**Comment by jseldess** [comment]
Sure, @mburke5678. Hoping to get back to this PR soon.

File: docs/tasks/configure-pod-container/projected-volume.md (hunk 1 above)
Comment (jseldess) [inline docs/tasks/configure-pod-container/projected-volume.md]: @jpeeler, I need some assistance. I can't seem to get the pod specification right. Here's what I did:

1. I created some secrets:

```
$ echo -n "admin" > ./username.txt
$ echo -n "1f2d1e2e67df" > ./password.txt
$ kubectl create secret generic user --from-file=./username.txt
$ kubectl create secret generic pass --from-file=./password.txt
```
2. I create a pod config:

```
apiVersion: v1
kind: Pod
metadata:
  name: test-projected-volume
spec:
  containers:
  - name: test-projected-volume
    image: busybox
    args:
    - sleep
    - "86400"
    volumeMounts:
    - name: all-in-one
      mountPath: "/projected-volume"
      readOnly: true
  volumes:
  - name: all-in-one
    projected:
      sources:
      - secret:
          name: user
      - secret:
          name: pass
```

3. I tried to create the pod:

```
kubectl create -f podspec.yaml
```
But I always get the following error:

```
error validating "podspec.yaml": error validating data: found invalid field projected for v1.Volume; if you choose to ignore these errors, turn validation off with --validate=false
```
I don't know what to do in response to `invalid field projected`. I don't see anything invalid. Can you help?

Comment (jpeeler) [inline docs/tasks/configure-pod-container/projected-volume.md]: @jseldess I'm not sure, I just tested it and it works fine for me. Is it possible you're running a version older than 1.6.0? The projected driver is somewhat new.

Comment (jseldess) [inline docs/tasks/configure-pod-container/projected-volume.md]: Indeed, I'm using an old version. I'll upgrade and try again. Thanks!

Comment (jseldess) [inline docs/tasks/configure-pod-container/projected-volume.md]: OK, I upgraded both minikube and kubectl and I know longer get that error. Hoping to wrap this up soon.

**Comment by jseldess** [comment]
@jpeeler, @chenopis, I think I've address all issues, but please let me know. Unfortunately, when fetching updates from upstream, I seem to have added all those commits to this PR. Happy to clean things up, if you could give me some guidance.

**Comment by jseldess** [comment]
Ugh. I didn't set myself up properly. Working off of my forked master branch. I'm going to close this PR and open a new PR with just the relevant updates. Apologies for my git stumbles.

**Comment by jpeeler** [comment]
FWIW, you can utilize force push (via git push --force-with-lease) to not have to open up a new PR.
